releng/release-tool/release-tool.go

```diff
@@ -0,0 +1,1186 @@
+	cacheDir := flag.String("cache-dir", "/tmp/release-tool", "The base directory used to cache git repos in")
+	cleanCacheDir := flag.Bool("clean-cache", true, "Clean the cache dir before executing")
+	githubTokenFile := flag.String("github-token-file", "", "file containing the github token.")
+	gitUser := flag.String("git-user", "", "git user")
+	gitEmail := flag.String("git-email", "", "git user email")
+	skipReleaseNotes := flag.Bool("skip-release-notes", false, "skip generating release notes for a tag")
+	force := flag.Bool("force", false, "force a release or release branch to occur despite blockers or other warnings")
+	promoteRC := flag.String("promote-release-candidate", "", "The tag of an rc release that will be promoted to an official release")
+
+	autoRelease := flag.Bool("auto-release", false, "Automatically perform branch cutting an releases based on time intervals")
+	autoReleaseCadance := flag.String("auto-release-cadance", "monthly", "set the auto release cadance to daily or monthly")
+	autoPromoteAfterDays := flag.Int("auto-promote-after-days", 7, "Set the set the time before autopromoting a release candidate")
+
+	flag.Parse()
+
+	if *org == "" {
+		log.Fatal("--org is a required argument")
+	} else if *repo == "" {
+		log.Fatal("--repo is a required argument")
+	} else if *gitUser == "" {
+		log.Fatal(" --git-user is required")
+	} else if *gitEmail == "" {
+		log.Fatal("--git-email is required")
+	} else if *githubTokenFile == "" {
+		log.Fatal("--github-token-file is a required argument")
+	}
+
+	tokenBytes, err := ioutil.ReadFile(*githubTokenFile)
+	if err != nil {
+		log.Fatal(fmt.Printf("ERROR accessing github token: %s ", err))
+	}
+	token := strings.TrimSpace(string(tokenBytes))
+
+	repoUrl := fmt.Sprintf("https://%s@github.com/%s/%s.git", token, *org, *repo)
+	infraUrl := fmt.Sprintf("https://%s@github.com/kubevirt/project-infra.git", token)
+	repoDir := fmt.Sprintf("%s/%s/https-%s", *cacheDir, *org, *repo)
+	infraDir := fmt.Sprintf("%s/%s/https-%s", *cacheDir, "kubevirt", "project-infra")
+
+	if *cleanCacheDir {
+		os.RemoveAll(repoDir)
+		os.RemoveAll(infraDir)
+	}
+
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	client := github.NewClient(tc)
+
+	r := releaseData{
+		repoDir:          repoDir,
+		infraDir:         infraDir,
+		repoUrl:          repoUrl,
+		infraUrl:         infraUrl,
+		repo:             *repo,
+		org:              *org,
+		newBranch:        *newBranch,
+		tag:              *releaseTag,
+		promoteRC:        *promoteRC,
+		skipReleaseNotes: *skipReleaseNotes,
+
+		gitUser:  *gitUser,
+		gitEmail: *gitEmail,
+		gitToken: token,
+
+		githubClient:    client,
+		githubTokenPath: *githubTokenFile,
+		dryRun:          *dryRun,
+		force:           *force,
+
+		blockerListCache: make(map[string]*blockerListCacheEntry),
+
+		now: time.Now(),
+	}
+
+	_, err = gitCommand("config", "--global", "user.name", r.gitUser)
```

Comment:
ah, interesting. so the tool actually is only meant to be run in a container but I can look into a per repo git config just in case someone does run it locally